plugins/module_utils/gandi_livedns_api.py

```diff
@@ -0,0 +1,264 @@
+        if zone_id:
+            url = '/zones/%s' % (zone_id)
+        else:
+            url = '/domains/%s' % (domain)
+
+        url += '/records'
+        if record:
+            url += '/%s' % (record)
+            if type:
+                url += '/%s' % (type)
+
+        records, status = self._gandi_api_call(url, error_on_404=False)
+
+        if status == 404:
+            return None
+
+        if not isinstance(records, list):
+            records = [records]
+
+        # filter by type if record is not set
+        if not record and type:
+            records = [r
+                       for r in records
+                       if r['rrset_type'] == type]
+
+        return records
+
+    def create_record(self, record, type, values, ttl, zone_id=None, domain=None):
+        if zone_id:
+            url = '/zones/%s' % (zone_id)
+        else:
+            url = '/domains/%s' % (domain)
+
+        url += '/records'
+        new_record = {
+            'rrset_name': record,
+            'rrset_type': type,
+            'rrset_values': values,
+            'rrset_ttl': ttl,
+        }
+        record, status = self._gandi_api_call(url, method='POST', payload=new_record)
+
+        if status in (201,):
+            return new_record
+
+        return None
+
+    def update_record(self, record, type, values, ttl, zone_id=None, domain=None):
+        if zone_id:
+            url = '/zones/%s' % (zone_id)
+        else:
+            url = '/domains/%s' % (domain)
+
+        url += '/records/%s/%s' % (record, type)
+        new_record = {
+            'rrset_values': values,
+            'rrset_ttl': ttl,
+        }
+        record, status = self._gandi_api_call(url, method='PUT', payload=new_record)
+        return record
+
+    def delete_record(self, record, type, zone_id=None, domain=None):
+        if zone_id:
+            url = '/zones/%s' % (zone_id)
+        else:
+            url = '/domains/%s' % (domain)
+        url += '/records/%s/%s' % (record, type)
+
+        cur_record, status = self._gandi_api_call(url, method='DELETE')
+
+    def delete_dns_record(self, record, type, ttl, values, zone=None, domain=None):
+
+        if type is None or record is None:
+            self.module.fail_json(msg="You must provide a type and a record to delete a record")
+
+        if zone:
+            zone_id = self._get_zone_id(zone)
+        else:
+            zone_id = None
+
+        records = self.get_records(record, type, zone_id=zone_id, domain=domain)
+
+        if records:
+            cur_record = records[0]
+
+            if ttl is not None and cur_record['rrset_ttl'] != ttl:
+                return None, self.changed
+            if values is not None and set(cur_record['rrset_values']) != set(values):
+                return None, self.changed
+
+            self.changed = True
+            if not self.module.check_mode:
+                result = self.delete_record(record, type, zone_id=zone_id, domain=domain)
```

Comment:
So, I may be missing something obvious, but if I want to remove 1 single record (let's say just 1 MX), I can't, as it seems that I would either remove all MX, or none, since value is not used anywhere in the delete_record method.

Reply:
Indeed, you can't remove a record's subset, you need to update it with its new value. (a PUT on {fqdn}/records/{rrset_name}/{rrset_type})

Reply:
Ack, specifying a value will either remove the record if it has only one value, or update the record with the new values (the remaining values)